coffee-chats/src/main/webapp/src/pages/GroupListPage.tsx

```diff
@@ -1,25 +1,57 @@
 import React from "react";
-import {Box, Container, Grid, TextField} from "@material-ui/core";
+import {Box, Container, Fab, Grid, Icon, TextField} from "@material-ui/core";
 import {GroupCard} from "../components/GroupCard";
+import {makeStyles} from "@material-ui/core/styles";
+import {GroupCreateDialog} from "../components/GroupCreateDialog";
+import {useFetch} from "../util/fetch";
+import {Group} from "../entity/Group";
+
+const useStyles = makeStyles((theme) => ({
+  extendedIcon: {
+    marginRight: theme.spacing(.5),
+  },
+  shrink: {
+    flexGrow: 0
+  }
+}));
 
 export function GroupListPage() {
+  const classes = useStyles();
+  const [groups, updateGroups]: [Group[], () => void] = useFetch("/api/groupList", []);
+  const [createDialogOpen, setCreateDialogOpen] = React.useState(false);
+
   return (
-      <Box mt={4}>
-        <Container maxWidth="md">
-          <Grid container spacing={2}>
-            <Grid item xs={12}>
-              <TextField
-                  fullWidth
-                  variant="outlined"
-                  label="Search your groups"/>
+      <React.Fragment>
+        <GroupCreateDialog
+            open={createDialogOpen}
+            setOpen={setCreateDialogOpen}
+            onSubmit={updateGroups} />
```

Comment:
Seems like an overcomplicated way to just call another fetch? Passing a fetch call into this prop would achieve the same goal.

Reply:
Can components by async functions? And even if yes, I'd have to resend the request every time the component is rendered. I think `useFetch` (which is a hook, remember) is actually a good solution here.